app/controllers/concerns/validate_move.rb

```diff
@@ -0,0 +1,15 @@
+module ValidateMove
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :validate_previous_id, only: :move
+  end
+
+  protected
+
+  def validate_previous_id
+    return true if params[:next_id].present? && params[:prev_id].blank?
+
+    head :unprocessable_entity unless controller_name.classify.constantize.exists? params[:prev_id]
```

Comment:
This line exceeds the [maximum line length](https://github.com/rubocop/ruby-style-guide#maximum-line-length). I suggest you configure your editor to show the 80 character limit.

It would also help the readability if you move `controller_name.classify.constantize` to a separate method.